src/engine/pieces/piece.ts

```diff
@@ -1,9 +1,14 @@
 import Player from '../player';
 import Board from '../board';
 import Square from '../square';
+import GameSettings from '../gameSettings';
 
 export default class Piece {
     public player: Player;
+    public static boardSize: number = GameSettings.BOARD_SIZE;
+    public isKing: Boolean = false;
+    public static diagonalSteps: Array<Array<number>> = [[-1, -1], [-1, 1], [1, -1], [1, 1]];
```

Comment:
```suggestion
    public static diagonalSteps: number[][] = [[-1, -1], [-1, 1], [1, -1], [1, 1]];
```

nitpick but it's more common practice to define arrays like this